### PR TITLE
Use new SniCompletionEvent from Netty

### PR DESCRIPTION
**Pull request by Sammers21**
A solution for #2226

### CODE REVIEW

File: src/main/java/io/vertx/core/net/impl/VertxSniHandler.java [hunk 1]
```diff
@@ -54,6 +51,23 @@ protected EventExecutor executor() {
         if (ctx == null) {
           throw new IllegalStateException();
         }
+        ctx.pipeline().addFirst(new ChannelInboundHandlerAdapter() {
```
Comment (Sammers21) [inline src/main/java/io/vertx/core/net/impl/VertxSniHandler.java]: Not sure is that correct place on the handler in the pipeline...

**Comment by Sammers21** [comment]
@vietj , should I actually submit a PR to https://github.com/vert-x3/vertx-dependencies with Netty version upgrade to _4.1.18.Final_

**Comment by vietj** [comment]
@Sammers21 no, I will take care of merging this one as I'll do a dependency upgrade early next week. Besides we should drop using vertx-dependencies in theory because otherwise we would need a CQ for it.

**Comment by vietj** [comment]
I'm planning a full dependency update early next week (cf vertx-dev list)

**Comment by vietj** [comment]
@Sammers21 can you try to drop the vertx sni handler class and do that using an event when the HTTP pipeline is assembled ?

**Comment by vietj** [comment]
@Sammers21 any news about this ? Netty 4.1.19.Final support has been merged in master

**Comment by Sammers21** [comment]
@vietj, great. But VertxSniHandler is also used in _NetServerImpl_, _NetSocketImpl_ .... if we would drop VertxSniHandler does it may cause code duplication. Is not it worth it to keep VertxSniHandler?

**Comment by vietj** [comment]
good point, let's keep it this way

**Comment by Sammers21** [comment]
@vietj, is something wrong?

**Comment by vietj** [comment]
yes it fails in CI, I had to revert.

**Comment by Sammers21** [comment]
where can I watch build results?

**Comment by Sammers21** [comment]
I would like to fix it

**Comment by vietj** [comment]
the build actually froze , I could reproduce on my machine, so just grab the master after I merged and test it

**Comment by Sammers21** [comment]
@vietj, I have set up Travis CI on my vertx fork. And when building the latest version of master... it fails... here is details: https://api.travis-ci.org/v3/job/326756484/log.txt . Is it expected?

**Comment by Sammers21** [comment]
and here is a link to the job: https://travis-ci.org/Sammers21/vert.x/jobs/326756484

**Comment by Sammers21** [comment]
while the PR job without your revert failed timeout: https://travis-ci.org/Sammers21/vert.x/jobs/326740521

**Comment by vietj** [comment]
wdym ?

**Comment by Sammers21** [comment]
I m trying to figure out where is the problem with my code. To begin with i m trying to ensure that all test passed without my changes made in this PR, but when i m running `mvn clean install` on latest version of master branch it [fails](https://api.travis-ci.org/v3/job/326756484/log.txt). I don't understand is it expected or not? Is my approach is right?

**Comment by Sammers21** [comment]
is it clear? :-)

**Comment by Sammers21** [comment]
But it works fine on my laptop.

**Comment by vietj** [comment]
I think your code is adding an handler but is not removing it, so I've mixed feelings about this approach after all because it's more complex than what I thought.

**Comment by Sammers21** [comment]
So now you think that it is better to leave it as is? And it is not worth it to make it works?

**Comment by Sammers21** [comment]
Or I should try one more time? :))

**Comment by vietj** [comment]
I think it's better to leave it as is, I would have tried and failed as well if that makes you more comfortable :-)

**Pull request merged**